--- pdfminer/layout.py	Wed Oct 07 15:22:55 2015
+++ pdfminer/layout.py	Fri Oct 16 10:05:09 2015
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 from utils import INF, Plane, get_bound, uniq, csort, fsplit
 from utils import bbox2str, matrix2str, apply_matrix_pt
-
+import logging
 
 ##  IndexAssigner
 ##
@@ -607,6 +607,9 @@
             y1 = max(obj1.y1, obj2.y1)
             objs = set(plane.find((x0, y0, x1, y1)))
             return objs.difference((obj1, obj2))
+        if len(boxes) > 100:
+            logging.warning("Too many boxes (%d) to group, skipping.", len(boxes))
+            return boxes
         # XXX this still takes O(n^2)  :(
         dists = []
         for i in xrange(len(boxes)):
